### PR TITLE
refactor: (GenAI) Reorganized Grounding Samples (Group B)

### PR DESCRIPTION
**Pull request by Thoughtseize1**
## Description
Created new folder

| Region_tag                                        | New file                    |
|---------------------------------------------------|-----------------------------|
| generativeaionvertexai_gemini_grounding_with_web  | grounding/web_example.py    |
| generativeaionvertexai_gemini_grounding_with_vais | grounding/vais_example.py   |
| generativeaionvertexai_sdk_grounding              | grounding/simple_example.py |

❗Files need to be deleted after approval in `/generative_ai` folder:

- `grounding.py`
- `grounding_test.py`
- `gemini_grounding_example.py`
- Marked tests in `test_gemini_examples.py`

**I was unable to successfully test the following Samples. PTAL at** [doc](https://docs.google.com/document/d/17wDsm9pJeyBkjRnvuJe7mmwC3kii2WJx6TbbfHobMzc/edit?usp=sharing) 

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: generative_ai/grounding/simple_example.py
Comment (holtskinner) [inline generative_ai/grounding/simple_example.py:1]: Maybe change the name to be `palm_example.py`

**Comment by Thoughtseize1** [comment]
I encountered several issues while testing these samples. In general:
`google.api_core.exceptions.InternalServerError: 500 Internal error encountered.`

Please refer to the [document](https://docs.google.com/document/d/17wDsm9pJeyBkjRnvuJe7mmwC3kii2WJx6TbbfHobMzc/edit?usp=sharing) where I have outlined the problems and possible improvements in the documentation.

**Review by holtskinner — DISMISSED** [review]

**Comment by msampathkumar** [comment]
Update: `google.api_core.exceptions.InternalServerError: 500 Internal error encountered.` - As the CICD tests run in parallel, this error seem occur. Manually re-starting the CICD.

File: generative_ai/grounding/simple_example.py
Comment (msampathkumar) [reply generative_ai/grounding/simple_example.py:1]: File is renamed. Marking this as closed.

**Review by msampathkumar — APPROVED** [review]

**Pull request merged**